main.cc

```diff
@@ -0,0 +1,446 @@
+#define CROW_USE_BOOST 1
+#include <crow_all.h>
+#include <stdio.h>
+
+#include <boost/program_options.hpp>
+#include <cctype>
+#include <ctime>
+#include <iostream>
+#include <memory>
+#include <optional>
+#include <pqxx/pqxx>
+#include <stdexcept>
+#include <regex>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include "functions.h"
+namespace po = boost::program_options;
+class InvalidArguments : std::exception {};
+using namespace std;
+auto get_args(int argc, char** argv) -> po::variables_map {
+  po::options_description desc("Allowed options");
+  desc.add_options()("help", "show help")("pgdsn", po::value<std::string>(),
+                                          "PostgreSQL DSN");
+  po::variables_map vm;
+  po::store(po::parse_command_line(argc, argv, desc), vm);
+  if (vm.count("help")) {
+    std::cout << desc << std::endl;
+    exit(0);
+  }
+  if (!vm.count("pgdsn")) {
+    throw InvalidArguments();
+  }
+  return vm;
+}
+
+auto establish_pgconnection(std::string_view dsn) -> pqxx::connection {
+  return pqxx::connection(dsn.data());
+}
+
+int main(int argc, char** argv) {
+  auto vm = get_args(argc, argv);
+  auto dsn = vm["pgdsn"].as<std::string>();
+
+  DB postgres("postgres", dsn);
+
+  crow::App<crow::CORSHandler> app;
+  auto& cors = app.get_middleware<crow::CORSHandler>();
+
+  // Добавить курьера
+  CROW_ROUTE(app, "/api/couriers")
+      .methods(crow::HTTPMethod::POST)([&postgres](const crow::request& req) {
+        // парсинг json
+        try {
+          if (req.method != crow::HTTPMethod::POST) {
+            return crow::response(405, "Method Not Allowed");
+          }
+
+          auto json = crow::json::load(req.body);
+          if (!json.has("age") || !json.has("gender") || !json.has("phone") ||
+              !json.has("active")) {
+            return crow::response(400, "Missing required fields");
+          }
+
+          int age = json["age"].i();
+          string gender = json["gender"].s();
+          string phone = json["phone"].s();
+          string active = json["active"].s();
+
+          db::courier courier1(1, age, gender, phone, active);
+
+          postgres.add_courier(courier1);
+          return crow::response(
+              201,
```

Comment:
Magic number. Сильно сомневаюсь что через неделю будет идеальное соображение что значит `201`

```suggestion
              crow::status::CREATED,
```

Список методов есть тут: http://crowcpp.org/master/reference/namespacecrow.html#aa3ae5c0994dff8425d7515e2806180d2